src/Integrations/Integrations/DatabaseIntegrationHelper.php

```diff
@@ -43,7 +43,10 @@ public static function injectDatabaseIntegrationData(HookData $hook, $backend, $
                 $propagationMode = \DDTrace\DBM_PROPAGATION_SERVICE;
             }
 
-            $query = self::propagateViaSqlComments($hook->args[$argNum], $hook->span()->service, $propagationMode);
+            $databaseService = in_array("peer.service", $hook->span()->meta)
+                ? $hook->span->meta["peer.service"] : $hook->span()->service;
```

Comment:
I'll defer to reasoning of the service to someone on the tracing team, but I think you want this:

```suggestion
            $databaseService = isset($hook->span->meta['peer.service'])
                ? $hook->span->meta['peer.service']
                : $hook->span()->service;
```

Reply:
Thanks for that! Is the reason performance or is there something else I'm missing?

Also, sorry for making you look at this ahead of time, is there a way you flag PRs in this repo as WIP other than draft PRs? Should I not be drafting PRs?